clients/pkg/promtail/targets/azurelog/targetmanager.go

```diff
@@ -0,0 +1,75 @@
+package azurelog
+
+import (
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/grafana/loki/clients/pkg/promtail/api"
+	"github.com/grafana/loki/clients/pkg/promtail/scrapeconfig"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/kafka"
+	"github.com/grafana/loki/clients/pkg/promtail/targets/target"
+)
+
+// TargetManager manages a series of kafka targets.
+type TargetManager struct {
+	logger        log.Logger
+	targetSyncers map[string]*kafka.TargetSyncer
+}
+
+// NewTargetManager creates a new Kafka managers.
+func NewTargetManager(
+	reg prometheus.Registerer,
+	logger log.Logger,
+	pushClient api.EntryHandler,
+	scrapeConfigs []scrapeconfig.Config,
+) (*TargetManager, error) {
+	tm := &TargetManager{
+		logger:        logger,
+		targetSyncers: make(map[string]*kafka.TargetSyncer),
+	}
+	for _, cfg := range scrapeConfigs {
+		t, err := NewSyncer(reg, logger, cfg, pushClient)
+		if err != nil {
+			return nil, err
+		}
+		tm.targetSyncers[cfg.JobName] = t
+	}
+
+	return tm, nil
+}
+
+// Ready returns true if at least one Kafka target is active.
+func (tm *TargetManager) Ready() bool {
+	for _, t := range tm.targetSyncers {
+		if len(t.ActiveTargets()) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func (tm *TargetManager) Stop() {
+	for _, t := range tm.targetSyncers {
+		if err := t.Stop(); err != nil {
+			level.Error(t.Logger()).Log("msg", "error stopping azurelog target", "err", err)
```

Comment:
Do we need to use the target's logger here? It doesn't look like there's anything special about it vs using `tm.logger`. This would also avoid the need for the `TargetSyncer` `Logger()` function all together.

Reply:
Good question @kgeckhart 👍. I think we do not. I did it the same way it was done in the Kafka target. Just to keep it consistent. Now I changed it in both palaces, as you suggested 😃